pos_automatic_cashdrawer/static/src/js/devices.js

```diff
@@ -0,0 +1,299 @@
+/*
+    POS Automatic Cashdrawer module for Odoo
+    Copyright (C) 2016 Aurélien DUMAINE
+    Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+    @author: Aurélien DUMAINE
+    The licence is in the file __manifest__.py
+*/
+
+odoo.define('pos_automatic_cashdrawer.devices', function (require) {
+    "use strict";
+
+    var devices = require('point_of_sale.devices');
+    var core = require('web.core');
+    var framework = require('web.framework');
+
+    var _t = core._t;
+
+    devices.ProxyDevice.include({
+
+        /*
+            Overload keepalive.
+            This function is called right after the PosBox is connected and the handshake is made.
+            Normally, it starts a keepalive timer that request /hw_proxy/status_json
+            But we're also sending the Cashdrawer connection config here.
+        */
+        keepalive: function () {
+            this._super.apply(this, arguments);
+            if (this.pos.config.iface_automatic_cashdrawer) {
+                this.message('cashlogy/connect', {
+                    'config': {
+                        'host': this.pos.config.iface_automatic_cashdrawer_ip_address,
+                        'port': this.pos.config.iface_automatic_cashdrawer_tcp_port
+                    }
+                });
+            }
+        },
```

Comment:
Unsure, but it seems an uncommon place to send the cashdrawer connection info. In standard, it only starts a loop that updates the connection status. [Here](https://github.com/OCA/pos/blob/12.0/pos_payment_terminal/static/src/js/devices.js#L50) a method is added to `devices.ProxyDevice` and called in [`screens.PaymentScreenWidget`](https://github.com/OCA/pos/blob/12.0/pos_payment_terminal/static/src/js/screens.js#L25) to send data